python/common/org/python/types/Float.java

```diff
@@ -14,6 +14,46 @@ public int hashCode() {
         return new java.lang.Double(this.value).hashCode();
     }
 
+    public org.python.types.Int __hash__() {
+        final long _PyHashINF = 314159L, _PyHashNAN = 0L, _PyHashMULTIPLIER = 1000003L;
```

Comment:
This hashing algorithm is complex, so I'm guessing it wasn't something you invented yourself - in which case, you should reference where the original came from. This is partially to give appropriate credit to whoever wrote the original, but also to assist whoever comes here later to work out what is going on.